content/thin-provisioning-redesign.md

```diff
@@ -0,0 +1,137 @@
++++
+title = "Thin provisioning redesign"
+date = 2022-05-18
+weight = 24
+template = "page.html"
+render = true
++++
+
+*John Baublitz, Stratis Team*
+
+Overview
+========
+For a while, we've bumped into a number of problems with our thin provisioning implementation
+around reliability and safety for users. After gathering a lot of feedback on our thin
+provisioning layer, we put together [a proposal] for improvements to how we currently handle
+allocations.
+
+The changes can largely be divided up into three areas of improvement:
+* Predictability
+* Safety
+* Reliability
+
+Predictability
+==============
+We made two notable changes to make behavior in the thin provisioning layer well-defined and
+predictable for users. Both parts relate to an existing thin provisioning tool,
+`thin_metadata_size`. This tool allows users to calculate the amount of metadata needed for
+a thin pool with a given size and number of thin devices (filesystems and snapshots in the
+case of stratisd). We have started taking advantage of `thin_metadata_size` to make our
+metadata space reservation more precise. Instead of our previous approach of allocating
+a fixed fraction of the available space, we now calculate the exact amount of space required for
+a given pool size and number of filesystems and snapshots. The second change is a switch
+to lazy allocation. Previously, we allocated greedily which meant that every time a device
+was added, it would allocate a certain amount of space for data and metadata regardless of
```

Comment:
Use passive voice? "it" has no referrent so "a certain amount of space would be allocated..." would make more sense.